### PR TITLE
Add two tests for checking if read() and recv() handle EOF on close/shutdown correctly.

### PR DESCRIPTION
**Pull request by sgielen**
I found out that this behaviour was incorrect on Cosix, even though all relevant tests were succeeding. There are existing tests for checking this behaviour with poll(), but none yet for a blocking read() and recv(). Perhaps we'll want to add symmetric tests for write() and send() as well.

Verified by re-introducing the same bug into Cosix and confirming that the tests do fail.

### CODE REVIEW

File: src/libc/sys/socket/shutdown_test.c [hunk 1]
```diff
@@ -52,3 +54,61 @@ TEST(shutdown, example) {
   ASSERT_EQ(0, close(fds[0]));
   ASSERT_EQ(0, close(fds[1]));
 }
+
+static void *sleep_then_shutdown_read(void *arg) {
+  // Sleep for a second.
+  struct timespec ts = {.tv_sec = 0, .tv_nsec = 100000000L};
```
Comment (EdSchouten) [inline src/libc/sys/socket/shutdown_test.c]: Could you make this less than one second? The problem with such long sleeps is that they make the unit test suite a lot slower.   Even if we would sleep too briefly, the test should deal with that correctly, right?

Just use 100 msec. Also no need to put `L` behind the integer literal. :-)

File: src/libc/sys/socket/shutdown_test.c [hunk 2]
```diff
@@ -52,3 +54,61 @@ TEST(shutdown, example) {
...
+  ASSERT_EQ(0, clock_nanosleep(CLOCK_MONOTONIC, 0, &ts));
+
+  // Shutdown the given file handle for reading.
+  ASSERT_EQ(0, shutdown(*(int*)arg, SHUT_RD));
```
Comment (EdSchouten) [inline src/libc/sys/socket/shutdown_test.c]: Can you please run `clang-format`?

File: src/libc/sys/socket/shutdown_test.c (hunk 1 above)
Comment (EdSchouten) [reply src/libc/sys/socket/shutdown_test.c]: Oh, wait. You are already using 100 msec. It's just that the comment is stale.

Comment (sgielen) [reply src/libc/sys/socket/shutdown_test.c]: Ah, oops! Update coming. If we sleep too briefly, the `recv()` / `read()` may not have started blocking yet, but that may at most just cause the test to succeed by accident. With 100 ms, that would be insignificantly rare I think.

**Comment by sgielen** [comment]
@EdSchouten patch is updated!

**Pull request merged**